src/shared/config/authentication.ts

```diff
@@ -55,6 +55,17 @@ if (castBooleanEnv('AUTH_GOOGLE_ENABLE')) {
     clientSecret: process.env.AUTH_GOOGLE_CLIENT_SECRET
   }
 }
+
+// Apple OAuth2 provider settings
+if (castBooleanEnv('AUTH_APPLE_ENABLE')) {
+  AUTH_PROVIDERS.apple = {
+    clientID: process.env.AUTH_APPLE_CLIENT_ID,
+    teamID: process.env.AUTH_APPLE_TEAM_ID,
+    keyID: process.env.AUTH_APPLE_KEY_ID,
+    key: process.env.AUTH_APPLE_PRIVATE_KEY,
+    callbackURL: process.env.AUTH_APPLE_CALLBACK_URL // optional
```

Comment:
1. Is `callbackURL` really optional?
2. Won't we cover this here: https://github.com/nhost/hasura-backend-plus/blob/v2/src/routes/auth/providers/utils.ts#L105?

Or is this some special case with Apple ID?

Reply:
I think we can remove this and this enforce the structured urls `/auth/providers/[]/callback` instead of complicating things.